keras_eval/eval.py

```diff
@@ -176,28 +175,29 @@ def save_probabilities_labels(self, id, save_path):
 
     def compute_inference_probabilities(self, concept_dictionary, probabilities):
         '''
-        Combines probabilities based on key "group" in concept_dictionary and saves the values in self.probabilities
+        Computes the probability inference based on key "group" in concept_dictionary and saves the values in self.probabilities
 
         Args:
             concept_dictionary: It is the dictionary which contains all the granular concepts and the mapping with the groups.
             probabilities: These are computed granular probabilities
 
         '''
+        self.combined_probabilities = utils.combine_probabilities(probabilities, self.combination_mode)
```

Comment:
This will be called again by `get_metrics` for the same set of probabilities, is there a way to avoid that? 🤔

Reply:
The previous step when calling `get_metrics` (or `compute_inference_probabilities`) is the private function `_compute_probabilities_generator`. 

I thought to add the `combine_probabilities` in there, but the problem is that we loose the sense if the next functions (`get_metrics` and `compute_inference_probabilities`) should work with `self.probabilities` or `self. combined_probs` depending if it's a single model or ensemble.